xdsl/backend/register_type.py

```diff
@@ -28,26 +28,32 @@ class RegisterType(ParametrizedAttribute, TypeAttribute, ABC):
     index: ParameterDef[IntAttr | NoneAttr]
     spelling: ParameterDef[StringAttr]
 
-    def __init__(self, spelling: str):
-        super().__init__(self._parameters_from_spelling(spelling))
+    def __init__(self, index: IntAttr | NoneAttr, spelling: StringAttr):
+        super().__init__((index, spelling))
 
     @classmethod
     def unallocated(cls) -> Self:
         """
         Returns an unallocated register of this type.
         """
-        return cls("")
+        return cls(NoneAttr(), StringAttr(""))
 
     @classmethod
     def _parameters_from_spelling(
-        cls, spelling: str
+        cls, spelling: StringAttr
     ) -> tuple[IntAttr | NoneAttr, StringAttr]:
         """
         Returns the parameter list required to construct a register instance from the given spelling.
         """
-        index = cls.abi_index_by_name().get(spelling)
+        index = cls.abi_index_by_name().get(spelling.data)
         index_attr = NoneAttr() if index is None else IntAttr(index)
-        return index_attr, StringAttr(spelling)
+        return index_attr, spelling
+
+    @classmethod
+    def from_spelling(cls, spelling: StringAttr | str) -> Self:
```

Comment:
This name has been bothering me in the last few PRs, because in my mind spelling redirects to orthography.
Moreover, there is no mention in the RISC-V or Arm ISA specification manuals, so it is completely foreign.

Can't we call this simply and plainly `from_name`? I mean, we also have `register_name` at some point, and it won't affect verification even if I give an erroneous register name.

Lastly, for example, in the RISC-V case you have the register number and the register name (e.g., `x8` and `s0`/`fp`).
The index will cover the register number part, while for `xN` this is just a mnemonic name of that, and the `sN` is the corresponding ABI name.

In any case, my point is that these are just names, and we should just call them that.

Reply:
Hopefully, this is just a find/replace kind of thing.

Reply:
I completely agree, with the caveat that `name` is already taken. Let's discuss this on Zulip.